pkg/trace/api/api.go

```diff
@@ -655,32 +641,11 @@ func (r *HTTPReceiver) watchdog(now time.Time) {
 			log.Criticalf("Killing process. Memory threshold exceeded: %.2fM / %.2fM", current/1024/1024, allowed/1024/1024)
 			killProcess("OOM")
 		}
-		rateMem = computeRateLimitingRate(r.conf.MaxMemory, float64(wi.Mem.Alloc), r.RateLimiter.RealRate())
-		if rateMem < 1 {
-			log.Warnf("Memory threshold exceeded (apm_config.max_memory: %.0f bytes): %d", r.conf.MaxMemory, wi.Mem.Alloc)
-		}
-	}
-	rateCPU := 1.0
-	if r.conf.MaxCPU > 0 {
-		if cpuErr != nil {
-			log.Errorf("Error retrieving current CPU usage: %v. Reusing previous value", cpuErr)
-		}
-		rateCPU = computeRateLimitingRate(r.conf.MaxCPU, wi.CPU.UserAvg, r.RateLimiter.RealRate())
-		if rateCPU < 1 {
-			log.Warnf("CPU threshold exceeded (apm_config.max_cpu_percent: %.0f): %.0f", r.conf.MaxCPU*100, wi.CPU.UserAvg*100)
-		}
 	}
-
-	r.RateLimiter.SetTargetRate(math.Min(rateCPU, rateMem))
-
-	stats := r.RateLimiter.Stats()
-
-	info.UpdateRateLimiter(*stats)
 	info.UpdateWatchdogInfo(wi)
 
 	metrics.Gauge("datadog.trace_agent.heap_alloc", float64(wi.Mem.Alloc), nil, 1)
 	metrics.Gauge("datadog.trace_agent.cpu_percent", wi.CPU.UserAvg*100, nil, 1)
-	metrics.Gauge("datadog.trace_agent.receiver.ratelimit", stats.TargetRate, nil, 1)
```

Comment:
We'll need to update https://docs.datadoghq.com/tracing/troubleshooting/agent_apm_metrics/

Reply:
https://github.com/DataDog/documentation/pull/19256